app/helpers/fhir_helper.rb

```diff
@@ -107,57 +112,67 @@ def races_as_hash(patient)
 
   # Create a hash of atttributes that corresponds to a Sara Alert Patient (and can be used to
   # create new ones, or update existing ones), using the given FHIR::Patient.
+  # Hash is of the form:
+  # {
+  #  attribute_name: { value: <converted-value>, path: <fhirpath-to-corresponding-fhir-element> }
+  # }
   def patient_from_fhir(patient, default_jurisdiction_id)
     symptom_onset = from_date_extension(patient, ['symptom-onset-date'])
     address = from_address_by_type_extension(patient, 'USA')
+    address_index = patient&.address&.index(address)
     foreign_address = from_address_by_type_extension(patient, 'Foreign')
+    foreign_address_index = patient&.address&.index(foreign_address)
+    primary_phone = patient&.telecom&.find { |t| t&.system == 'phone' }
+    secondary_phone = patient&.telecom&.select { |t| t&.system == 'phone' }&.second
+    email = patient&.telecom&.find { |t| t&.system == 'email' }
```

Comment:
Something nefarious can happen here with the `PrimaryContactValidator`. Let's say the `preferred_contact_method` is `Telephone call`, and neither a `primary_phone` or `secondary_phone` is given. This is an error that comes from the `PrimaryContactValidator` that is placed on the `primary_telephone` element. The `path` on this attribute is `Patient.telecom[].value` since no index is found, but then the error message you get looks like this: 
```
Value '7848287919fake@example.com' for 'Primary Telephone' is required when 'Primary Contact Method' is 'Telephone call'
```
Which is a bit confusing. I think the `FHIRPath` is just evaluating as best as it can, and so it finds an email (if one exists) in the `telecom` array. This is a similar case to the other `address_index`. One possible approach would just be to change the `PrimaryContactValidator` to add an error on `:base`. I've been thinking for a bit that this error really belongs on `:base` anyway, since it isn't necessarily an issue with `primary_telephone`, but rather an issue with how `primary_telephone` and `preferred_contact_method` combine. But if we put the error on `:base` that raises the question of what the FHIRPath should be. I think in the case of an error like this, I wouldn't mind just not having FHIRPath, since the error isn't really with one specific value we can point to.

Reply:
I wouldn't call that nefarious, but yes this is definitely something that needs to be fixed. In a perfect world in the example you gave I think the FHIRPath `Patient.telecom` would make the most sense for the expression (though not to include in the diagnostics message string). However, I'm frankly just not seeing a good way to get that result, without some major refactor that splits the FHIRPath construction logic between the FHIR helper and the Patient model/validators, and that juice is not worth that squeeze. I'm not super happy with this, but I think I agree with your statement that not having a FHIRPath in the case of `:base` validation errors is acceptable. I'll make that change.

Reply:
Done https://github.com/SaraAlert/SaraAlert/pull/743/commits/102a1e26616034b64c512578fb44ae9c7076f8a9